components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -216,52 +284,15 @@ private void sendToConsentUri(HttpServletRequest request, HttpServletResponse re
-        String state = request
-                .getParameter(OIDCSessionConstants.OIDC_STATE_PARAM);
         String idTokenHint = request.getParameter(OIDCSessionConstants.OIDC_ID_TOKEN_HINT_PARAM);
 
         if (idTokenHint != null) {
-
-            try {
-                if (!validateIdToken(idTokenHint)) {
-                    String msg = "Signature validation failed for id token.";
-                    redirectURL = OIDCSessionManagementUtil
-                            .getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
-                    response.sendRedirect(redirectURL);
-                    return;
-                }
-
-                String clientId = extractClientFromIdToken(idTokenHint);
-                OAuthAppDAO appDAO = new OAuthAppDAO();
-                OAuthAppDO oAuthAppDO = appDAO.getAppInformation(clientId);
-
-                if (!validatePostLogoutUri(postLogoutRedirectUri, oAuthAppDO.getCallbackUrl())) {
-                    String msg = "Post logout URI not does not match with registered callback URI.";
-                    redirectURL = OIDCSessionManagementUtil
-                            .getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
-                    response.sendRedirect(redirectURL);
-                    return;
-                }
-                Map<String, String> paramMap = new HashMap<>();
-                paramMap.put(OIDCSessionConstants.OIDC_CACHE_CLIENT_ID_PARAM, clientId);
-                OIDCSessionDataCacheEntry cacheEntry = new OIDCSessionDataCacheEntry();
-                cacheEntry.setIdToken(idTokenHint);
-                cacheEntry.setPostLogoutRedirectUri(postLogoutRedirectUri);
-                cacheEntry.setState(state);
-                cacheEntry.setParamMap(new ConcurrentHashMap<>(paramMap));
-
-                Cookie opBrowserStateCookie = OIDCSessionManagementUtil.getOPBrowserStateCookie(request);
-                addSessionDataToCache(opBrowserStateCookie.getValue(), cacheEntry);
-                redirectURL = OIDCSessionManagementUtil.getOIDCLogoutConsentURL();
-            } catch (ParseException e) {
-                String msg = "No valid session found for the received session state.";
-                redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
-            } catch (InvalidOAuthClientException | IdentityOAuth2Exception e) {
-                String msg = "Error occurred while getting application information. Client id not found";
-                redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
+            redirectURL = processLogoutRequest(request, response);
+            if (!StringUtils.isEmpty(redirectURL)) {
+                response.sendRedirect(redirectURL);
+                return;
             }
+            redirectURL = OIDCSessionManagementUtil.getOIDCLogoutConsentURL();
```

Comment:
Move this to outside to if block
`redirectURL = OIDCSessionManagementUtil.getOIDCLogoutConsentURL();`